lib/backend/s3backend/client.go

```diff
@@ -239,49 +252,24 @@ func (c *Client) List(prefix string, opts ...backend.ListOption) (*backend.ListR
 			}
 			names = append(names, name)
 		}
-	}
-
-	if options.Paginated {
-		listInput := &s3.ListObjectsV2Input{
-			Bucket:  aws.String(c.config.Bucket),
-			MaxKeys: aws.Int64(int64(options.MaxKeys)),
-			Prefix:  aws.String(path.Join(c.pather.BasePath(), prefix)[1:]),
-		}
-
-		if options.ContinuationToken != "" {
-			listInput.ContinuationToken = aws.String(options.ContinuationToken)
-		}
 
-		listOutput, err := c.s3.ListObjectsV2(listInput)
-		if err != nil {
-			return nil, err
+		if page.IsTruncated != nil && *page.IsTruncated && page.NextContinuationToken != nil {
```

Comment:
would there be a case when `nextContinuationToken` is set when pagination is disabled? why
```
if !options.Paginated {
		nextContinuationToken = ""
	}
```

Reply:
If pagination is disabled and `numKeysWithPrefix > ListMaxKeys > len(ListObjectsV2Output.Contents)`, then we will have len(names) == ListMaxKeys and a continuation token in nextContinuationToken, so we need to mask it out.

Alternatively I can add options.Paginated to the condition to capture the token.